scene/resources/compressed_texture.cpp

```diff
@@ -155,6 +155,8 @@ Error CompressedTexture2D::load(const String &p_path) {
 	path_to_file = p_path;
 	format = image->get_format();
 
+	RS::get_singleton()->texture_set_path(texture, get_path());
```

Comment:
After the automatic reimport caused by detection, the texture path in RS was not being set, so the roughness texture detection, which relies on the normal texture's path, failed.

Reimport doesn't cause a call to `Resource.set_path()` because the path doesn't change, however this state is invalidated. It should makes sense then to set it here, probably for all texture types, not just `CompressedTexture2D`, unless there is somewhere better.